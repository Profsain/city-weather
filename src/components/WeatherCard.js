import React from 'react';
import { useSelector } from 'react-redux';
import { Grid } from '@mui/material';
import SingleCard from './SingleCard';
import setWeatherIcon from '../helper/setWeatherIcons';
import '../styles/WeatherCard.css';

const WeatherCard = () => {
  const nextFiveData = useSelector((state) => state.weather.nextFiveDaysData);

  return (
    <div className="WeatherCard">
      <Grid
        container
        gap={3}
        direction="row"
        justifyContent="center"
        alignItems="center"
      >
        {nextFiveData.list.map((weather) => (
          <Grid item xs={4} sm={2} key={weather.weather.id}>
            <SingleCard
              condition={weather.weather[0].main}
              temp={weather.main.temp}
              humidity={weather.main.humidity}
              icon={setWeatherIcon(weather.weather[0].main)}
            />
          </Grid>

        ))}
      </Grid>
    </div>
  );
};

export default WeatherCard;
