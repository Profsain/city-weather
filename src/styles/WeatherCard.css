.WeatherCard {
  padding-top: 18px;
}