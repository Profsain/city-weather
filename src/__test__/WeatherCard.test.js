import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configurestore';
import WeatherCard from '../components/WeatherCard';

it('Renders WeatherCard component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <WeatherCard />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
