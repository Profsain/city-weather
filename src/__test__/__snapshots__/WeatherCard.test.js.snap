// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders WeatherCard component correctly 1`] = `
<div
  className="WeatherCard"
>
  <div
    className="MuiGrid-root MuiGrid-container css-15mc7ki-MuiGrid-root"
  />
</div>
`;
